integrations/nlu-snips/src/SnipsNlu.ts

```diff
@@ -0,0 +1,191 @@
+import { join as joinPaths, resolve } from 'path';
+import { v4 as uuidV4 } from 'uuid';
+import {
+  axios,
+  AxiosRequestConfig,
+  AxiosResponse,
+  DynamicEntitiesMode,
+  Extensible,
+  HandleRequest,
+  Jovo,
+  JovoError,
+  JovoRequest,
+  NluData,
+  NluPlugin,
+  OutputTemplate,
+} from '@jovotech/framework';
+import { InputType, Intent, JovoModelData } from 'jovo-model';
+import { SnipsNluConfig, SnipsNluResponse } from './interfaces';
+
+export class SnipsNlu extends NluPlugin<SnipsNluConfig> {
+  install(parent: Extensible): Promise<void> | void {
+    super.install(parent);
+
+    parent.middlewareCollection.use('$output', this.generateDynamicEntities.bind(this));
+  }
+
+  getDefaultConfig(): SnipsNluConfig {
+    return {
+      serverUrl: 'http://localhost:5000/',
+      modelsDirectory: 'models',
+      fallbackLanguage: 'en',
+      serverPath: '',
+      engineId: uuidV4(),
+    };
+  }
+
+  async process(handleRequest: HandleRequest, jovo: Jovo): Promise<NluData | undefined> {
+    const text: string | undefined = jovo.$request.getRawText();
+
+    if (!text) {
+      return;
+    }
+
+    const config: AxiosRequestConfig = {
+      url: '/engine/parse',
+      params: {
+        locale: this.getLocale(jovo.$request).substring(0, 2),
+        engine_id: this.config.engineId,
+        session_id: jovo.$session.id!,
+      },
+      data: { text },
+    };
+    const snipsNluResponse: SnipsNluResponse = await this.sendToSnips(config);
+    const nluData: NluData = {};
+    if (snipsNluResponse.intent.intentName) {
+      nluData.intent = { name: snipsNluResponse.intent.intentName };
+    }
+
+    for (const slot of snipsNluResponse.slots) {
+      if (!nluData.entities) {
+        nluData.entities = {};
+      }
+
+      // TODO: Why is this a map when we do have to provide the name in the object itself?
+      nluData.entities[slot.slotName] = { name: slot.slotName, value: slot.value.value };
+    }
+
+    return nluData;
+  }
+
+  private async generateDynamicEntities(handleRequest: HandleRequest, jovo: Jovo): Promise<void> {
+    const outputs: OutputTemplate[] = Array.isArray(jovo.$output) ? jovo.$output : [jovo.$output];
+    const locale: string = this.getLocale(jovo.$request);
+
+    for (const output of outputs) {
+      const listen = output.platforms?.[jovo.$platform.constructor.name]?.listen ?? output.listen;
+
+      if (typeof listen !== 'object' || !listen.entities) {
+        return;
+      }
+
+      if (listen.entities.mode === DynamicEntitiesMode.Clear) {
+        return;
+      }
+
+      for (const entity of listen.entities.types || []) {
+        const requestData: JovoModelData = { invocation: '', intents: [], inputTypes: [] };
+
+        // Get intent and values from Jovo model
+        // TODO: Read model location from project configuration instead of passing it to config again?
```

Comment:
Could this lead to a problem with webpack? Should we import the models files instead?

Reply:
Yes, sorry, you mentioned that already in our brainstorm session. I'll play around with importing the model files directly and give you feedback once I have a result.

Reply:
I have settled on an approach to either let the user specify the model path and bundle the models with the code, or to directly import the models into the config. I'll adjust the docs accordingly.